projects/extension/sql/idempotent/014-parsing.sql

```diff
@@ -0,0 +1,113 @@
+-------------------------------------------------------------------------------
+-- parsing_auto
+create or replace function ai.parsing_auto() returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'auto'
+    , 'config_type': 'parsing'
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- parsing_none
+create or replace function ai.parsing_none() returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'none'
+    , 'config_type': 'parsing'
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- parser_pymupdf
+create or replace function ai.parsing_pymupdf() returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'pymupdf'
+    , 'config_type': 'parsing'
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- parser_docling
+create or replace function ai.parsing_docling() returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'docling'
+    , 'config_type': 'parsing'
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- _validate_parsing
+create or replace function ai._validate_parsing
+( config pg_catalog.jsonb  -- has to contain both loading and parsing config
```

Comment:
I would create distinct parameters for each element you need. This makes it more clear what the function expects and can better enforce that everything needed is provided and in the right types.